src/hardware/hfd_api.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2020 UBports foundation
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Erfan Abdi <erfangplus@gmail.com>
+ */
+
+// C APIs
+#include <hfd/hardware/light.h>
+#include <hfd/hardware/vibrator.h>
+#include <hardware/android_hw_module.h>
+
+// Light
+IMPLEMENT_FUNCTION0(
```

Comment:
I yet have to test that, but I think it should be IMPLEMENT_CTOR0 instead, which allows to return nullptr if implementation part that is linked against bionic (inside Android image) is missing.